rcl/src/rcl/client.c

```diff
@@ -58,13 +58,7 @@ rcl_client_init(
   // check the options and allocator first, so the allocator can be passed to errors
   RCL_CHECK_ARGUMENT_FOR_NULL(options, RCL_RET_INVALID_ARGUMENT, rcl_get_default_allocator());
   const rcl_allocator_t * allocator = &options->allocator;
-  RCL_CHECK_FOR_NULL_WITH_MSG(
-    allocator->allocate, "allocate not set",
-    return RCL_RET_INVALID_ARGUMENT, rcl_get_default_allocator());
-  RCL_CHECK_FOR_NULL_WITH_MSG(
-    allocator->deallocate, "deallocate not set",
-    return RCL_RET_INVALID_ARGUMENT, rcl_get_default_allocator());
-
+  RCL_CHECK_ALLOCATOR(allocator, return RCL_RET_INVALID_ARGUMENT);
```

Comment:
This won't set the error message if the allocator is invalid. That's why the changes in https://github.com/ros2/rcl/pull/178
 caused the tests to fail, because they're [expecting the error message to be set](https://github.com/ros2/rcl/blob/72bcb6b5b648b75f0c2096607c9e2fe0266352c7/rcl/test/rcl/test_node.cpp#L346). Using `RCL_CHECK_ALLOCATOR_WITH_MSG` is closer to the existing behaviour.